pkg/policy/prefilter.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os/exec"
+	"path"
+	"sync"
+	"syscall"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/maps/cidrmap"
+)
+
+type preFilterMapType int
+
+const (
+	prefixesV4Dyn preFilterMapType = iota
+	prefixesV4Fix
+	prefixesV6Dyn
+	prefixesV6Fix
+	mapCount
+)
+
+const (
+	maxLKeys = 1024 * 64
+	maxHKeys = 1024 * 1024 * 20
+)
+
+type preFilterMaps [mapCount]*cidrmap.CIDRMap
+
+type preFilterConfig struct {
+	dyn4Enabled bool
+	dyn6Enabled bool
+	fix4Enabled bool
+	fix6Enabled bool
+}
+
+// PreFilter holds global info on related CIDR maps participating in prefilter
+type PreFilter struct {
+	Maps     preFilterMaps
+	Config   preFilterConfig
+	Revision int64
+	Mutex    sync.RWMutex
+}
+
+// WriteConfig dumps the configuration for the corresponding header file
+func (p *PreFilter) WriteConfig(fw io.Writer) {
+	p.Mutex.Lock()
+	defer p.Mutex.Unlock()
+
+	fmt.Fprintf(fw, "#define CIDR4_HMAP_ELEMS %d\n", maxHKeys)
+	fmt.Fprintf(fw, "#define CIDR4_LMAP_ELEMS %d\n", maxLKeys)
+
+	fmt.Fprintf(fw, "#define CIDR4_HMAP_NAME %s\n", path.Base(p.Maps[prefixesV4Fix].String()))
+	fmt.Fprintf(fw, "#define CIDR4_LMAP_NAME %s\n", path.Base(p.Maps[prefixesV4Dyn].String()))
+	fmt.Fprintf(fw, "#define CIDR6_HMAP_NAME %s\n", path.Base(p.Maps[prefixesV6Fix].String()))
+	fmt.Fprintf(fw, "#define CIDR6_LMAP_NAME %s\n", path.Base(p.Maps[prefixesV6Dyn].String()))
+
+	if p.Config.fix4Enabled {
+		fmt.Fprintf(fw, "#define CIDR4_FILTER\n")
+		if p.Config.dyn4Enabled {
+			fmt.Fprintf(fw, "#define CIDR4_LPM_PREFILTER\n")
+		}
+	}
+	if p.Config.fix6Enabled {
+		fmt.Fprintf(fw, "#define CIDR6_FILTER\n")
+		if p.Config.dyn6Enabled {
+			fmt.Fprintf(fw, "#define CIDR6_LPM_PREFILTER\n")
+		}
+	}
+}
+
+func (p *PreFilter) dumpOneMap(which preFilterMapType, to []string) []string {
```

Comment:
Same thing goes with what I wrote in CIDRDump wrt the `to`.

Reply:
See comment there.